New Tutorial: Introduction to libp2p (#574)
ProtoSchool/protoschool.github.io: src/tutorials/0009-introduction-to-libp2p/01.md

---
    title: "Why do we need libp2p?"
    type: "multiple-choice"
---

<div class="flex justify-center items-start mt4 mb4">
    <img class="w-34-ns w-75" src="/tutorial-assets/T0009L01-libp2p-logo.svg" alt="libp2p logo" />
</div>

> **[libp2p](https://libp2p.io) is modular network stack** that evolved out of [IPFS](https://ipfs.io) to become an independent project.

Reviewer: ```suggestion
**[libp2p](https://libp2p.io) is a modular network stack** that evolved out of [IPFS](https://ipfs.io) to become an independent project.
```

> In order to explain why libp2p is such an important piece of the decentralized web, we need to take a few steps back and understand where it came from, in its initial implementation inside of IPFS, a peer-to-peer file sharing system. Let's start by exploring the networking problems that IPFS was built to tackle.

Reviewer: ```suggestion
In order to explain why libp2p is such an important piece of the decentralized web, we need to take a few steps back and understand where it came from. libp2p's initial implementation began inside of IPFS, a peer-to-peer file sharing system. Let's start by exploring the networking problems that IPFS was built to tackle.
```

This reads a bit clearer to me as separate sentences.

There are so many things that can go wrong in this scenario, and the truth is, you probably didn't want the photo to be stored in some far away database which you can't delete. You just wanted to share your photo directly with your friend.

## Location Addressing vs Content Addressing

A lot of the issues that we face come from using a [location addressing](/content-addressing/02) approach to find the content the user is looking for, and serving it from that address. For example, we expect our libp2p logo to be available at [https://proto.school/tutorial-assets/T0009L01-libp2p-logo.svg](/tutorial-assets/T0009L01-libp2p-logo.svg), but what if the server of this website is down? What if the DNS server is down? What if the image was converted to a PNG file and so the new location is `T0009L01-libp2p-logo.png`? What if you are in a country that has blocked the `proto.school` domain?
All of these issues are common ones and are caused by location addressing, where we try to find the content we need by its location.

> [IPFS](https://ipfs.io), however, approaches this problem in a different way by using [content addressing](/content-addressing/03) instead. If we know the
> [CID](/anatomy-of-a-cid) (Content IDentifier) of our libp2p logo instaed of its location, we can simply ask the network who has the image with this CID.

Author: ```suggestion
[IPFS](https://ipfs.io), however, approaches this problem in a different way by using [content addressing](/content-addressing/03) instead. If we know the [CID](/anatomy-of-a-cid) (Content IDentifier) of our libp2p logo instead of its location, we can simply ask the network who has the image with this CID.
```
